File: Serene/Serene.Script/Imports/ServerImports/Northwind.TerritoryForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class TerritoryForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Territory";

        public TerritoryForm(string idPrefix) : base(idPrefix) {}

        public StringEditor TerritoryID { [InlineCode("{this}.w('TerritoryID', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor TerritoryDescription { [InlineCode("{this}.w('TerritoryDescription', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public LookupEditor RegionID { [InlineCode("{this}.w('RegionID', Serenity.CodeGeneration.ExternalType)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.TerritoryForm.ts
namespace Serene.Northwind {
    export interface TerritoryForm {
        TerritoryID: Serenity.StringEditor;
        TerritoryDescription: Serenity.StringEditor;
        RegionID: Serenity.LookupEditor;
    }

    export class TerritoryForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Territory';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!TerritoryForm.init)  {
                TerritoryForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;
                var w1 = s.LookupEditor;

                Q.initFormType(TerritoryForm, [
                    'TerritoryID', w0,
                    'TerritoryDescription', w0,
                    'RegionID', w1
                ]);
            }
        }
    }
}
